torchdrivesim/behavior/iai.py

```diff
@@ -30,19 +32,84 @@ def iai_initialize(location, agent_count, center=(0, 0), traffic_light_state_his
     return agent_attributes, agent_states, response.recurrent_states
 
 
-def iai_drive(location: str, agent_states: Tensor, agent_attributes: Tensor, recurrent_states: List,
-              traffic_lights_states: Optional[dict] = None):
+def iai_conditional_initialize(location, agent_count, agent_attributes=None, agent_states=None, recurrent_states=None, center=(0, 0), traffic_light_state_history=None):
     import invertedai
-    from invertedai.common import AgentState, AgentAttributes, Point
-    agent_attributes = [AgentAttributes(length=at[0], width=at[1], rear_axis_offset=at[2]) for at in agent_attributes]
-    agent_states = [AgentState(center=Point(x=st[0], y=st[1]), orientation=st[2], speed=st[3]) for st in agent_states]
-    response = invertedai.api.drive(
-        location=location, agent_states=agent_states, agent_attributes=agent_attributes,
-        recurrent_states=recurrent_states, traffic_lights_states=traffic_lights_states
+    try:
+        INITIALIZE_FOV = 120
+        conditional_agent_attributes = []
+        conditional_agent_states = []
+        conditional_recurrent_states = []
+        outside_agent_states = []
+        outside_agent_attributes = []
+        outside_recurrent_states = []
+
+        for i in range(len(agent_states)):
+            agent_state = agent_states[i]
+            dist = math.dist(center, (agent_state.center.x, agent_state.center.y))
+            if dist < INITIALIZE_FOV:
+                conditional_agent_states.append(agent_state)
+                conditional_agent_attributes.append(agent_attributes[i])
+                conditional_recurrent_states.append(recurrent_states[i])
+            else:
+                outside_agent_states.append(agent_state)
+                outside_agent_attributes.append(agent_attributes[i])
+                outside_recurrent_states.append(recurrent_states[i])
+
+        agent_count -= len(conditional_agent_states)
+        if agent_count > 0:
+            try:
+                seed = random.randint(1, 10000)
+                response = invertedai.api.initialize(
+                    location=location,
+                    agent_attributes=conditional_agent_attributes,
+                    states_history=[conditional_agent_states],
+                    agent_count=agent_count,
+                    location_of_interest=center,
+                    traffic_light_state_history=traffic_light_state_history,
+                    random_seed = seed
+                )
+                agent_attribute_list = response.agent_attributes + outside_agent_attributes
+                agent_state_list = response.agent_states + outside_agent_states
+                recurrent_state_list = response.recurrent_states + outside_recurrent_states
+            except Exception:
```

Comment:
This is overly broad and will capture things like keyboard interrupts and out of memory exceptions, which we don't want. You probably want to catch `InvertedAIError` and maybe a few others?